src/layers/domain/ods/__init__.py

```diff
@@ -0,0 +1,79 @@
+import time
+from functools import wraps
+from http import HTTPStatus
+from typing import Callable, ParamSpec, TypeVar
+
+import requests
+from requests.exceptions import RequestException
+
+ODS_API_ENDPOINT = (
+    "https://directory.spineservices.nhs.uk/ORD/2-0-0/organisations/{ods_code}"
+)
+BACKOFF_BASE_SECONDS = 2
+
+
+class UnexpectedStatusCode(Exception):
+    pass
+
+
+class OdsApiOffline(ExceptionGroup):
```

Comment:
Are we sure the only way we can get errors is if the ODS API is offline? at first glance I assumed it had an offline version of the data to hit reading this - do we want something of a naming convention for the exception groups to make it more obvious or would that be overthinking it?

Reply:
> Are we sure the only way we can get errors is if the ODS API is offline? 
Yes - any other error will lead us to raise our own generic internal server error, and `OdsApiOffline` can only be raised if there are 3 consecutive internal server errors on the ODS API

> at first glance I assumed it had an offline version of the data to hit reading this - do we want something of a naming convention for the exception groups to make it more obvious or would that be overthinking it?
could rename to `OdsApiOfflineError`?